external/audiocodec/streaming/rb.c

```diff
@@ -132,45 +130,40 @@ size_t rb_read_ext(rb_p rbp, void *ptr, size_t len, size_t offset)
 	size_t used = rb_used(rbp);
 	RETURN_VAL_IF_FAIL(offset < used, 0);
 
-	if (len > used - offset)
-		len = used - offset;
-	RETURN_VAL_IF_FAIL(len != 0, 0);
+	len = MINIMUM(len, (used - offset));
+	RETURN_VAL_IF_FAIL((len != 0), 0);
 
-	if (ptr != NULL)
-	{
+	if (ptr != NULL) {
 		// increase temp rd_idx
 		size_t rd_idx = rbp->rd_idx;
 		_incr(rbp, &rd_idx, offset);
 
 		// start to copy data
-	    rd_idx = (rd_idx & IDX_MASK);
+		rd_idx = (rd_idx & IDX_MASK);
 		size_t len_contiguous = rbp->depth - rd_idx;
 
-		if (len > len_contiguous)
-		{
-			memcpy(ptr, (void*)((uint8_t*)rbp->buf + rd_idx), len_contiguous);
-			memcpy((void*)((uint8_t*)ptr + len_contiguous), rbp->buf, (len - len_contiguous));
+		if (len > len_contiguous) {
+			memcpy(ptr, (void *)((uint8_t *) rbp->buf + rd_idx), len_contiguous);
+			memcpy((void *)((uint8_t *) ptr + len_contiguous), rbp->buf, (len - len_contiguous));
+		} else {
+			memcpy(ptr, (void *)((uint8_t *) rbp->buf + rd_idx), len);
 		}
-		else
-			memcpy(ptr, (void*)((uint8_t*)rbp->buf + rd_idx), len);
 	}
 
-    return len;
+	return len;
```

Comment:
It always returns len even if ptr is null, is it correct?

Reply:
Yes, it's correct, now I added some comments in code.